### PR TITLE
Remove Span.Status

### PR DESCRIPTION
**Pull request by iNikem**
Addresses open-telemetry/opentelemetry-specification#706

### CODE REVIEW

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [inline text/0134-remove-span-status.md:25]:
```suggestion
Several areas are affected by this change and will require a coordinated effort to make this happen.
```

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [inline text/0134-remove-span-status.md:70]: Since the field is called `Status.message` I suggest:
```suggestion
|Message *(optional)* | `otel.deprecated_status_message` | `{message}` |
```

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [inline text/0134-remove-span-status.md:69]: What value do we set if it is 'OK'? Do we omit the `otel.deprecated_status_code` attribute?

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [inline text/0134-remove-span-status.md:86]: I am not quite sure what happens if Status is removed but we do not (yet) have the "error"-like attribute. How will the errors be indicated? Do we lose this capability? Or the intent is to wait until "error" indicator (whatever it is) is first added to the specification and only after that we remove Status?

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:69]: Yes, that was my idea. And probably should skip setting `otel.deprecated_status_message` as well

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:86]: Current proposal is to remove status right now, file a follow up issue and work with Errors WG to resolve it. Either via one of the already existing proposals, like open-telemetry/opentelemetry-specification#822, or a new one.

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [reply text/0134-remove-span-status.md:69]: Sounds good, please make it explicit in the text.

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [reply text/0134-remove-span-status.md:86]: Got it. I see in this OTEP that you already thought through the transition process, which is great, but I need to mention this for visibility: we must be careful with Collector changes. There are people who use OpenTelemetry Collector in production already. All intermediate states of the Collector need to be fully functional.

**Comment by iNikem** [comment]
@open-telemetry/technical-committee Can this be merged now? Then I can proceed doing actual changes.

File: text/0134-remove-span-status.md
Comment (Oberon00) [inline text/0134-remove-span-status.md:43]: Should the OTLP exporter (optionally) do the translation of grpc and HTTP status attributes to deprecated_status for now? Or is this not worth it?

File: text/0134-remove-span-status.md
Comment (Oberon00) [inline text/0134-remove-span-status.md:66]: Is the previous point not enough? Do we need to duplicate the status as attributes?

File: text/0134-remove-span-status.md
Comment (Oberon00) [inline text/0134-remove-span-status.md:65]: Should a non-empty message also be translated independently of the status code?

File: text/0134-remove-span-status.md
Comment (Oberon00) [inline text/0134-remove-span-status.md:80]:
```suggestion
Jaeger `error` tag will be translated in the usual way to an OTLP error attribute
(see [Open questions](#open-questions)).
```

The use of code formatting was confusing me when reading this, because that made it sound like the attribute actually exists with that name.

File: text/0134-remove-span-status.md
Comment (Oberon00) [inline text/0134-remove-span-status.md:81]:
```suggestion
* When exporting data into Jaeger format, if a new temporary attribute `otel.deprecated_status_code` is present
and does indicate a non-OK status,
```

**Review by Oberon00 — APPROVED** [review]
I have some questions about details, but all in all I agree to remove Status.

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:43]: As one of the reasons to remove `status` was the ambiguity of such translations (at least in http part), I don't think it is worth doing. And ideally we will not produce this Google codes based status anymore.

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:66]: Yes, because new OTLP receivers will not read `status` field from the wire. But we still may want to pass this information if it was available in the Collector.

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:65]: Message is part of the `Status`, how can it be independent?

Comment (Oberon00) [reply text/0134-remove-span-status.md:65]: I mean, it makes sense to add deprecated_status when "incoming OTLP data has `Span.Status` present and it is not equal to `OK`", but it may make sense to independently set deprecated_message when "incoming OTLP data has `Span.Status` present and **message non-empty**`"

File: text/0134-remove-span-status.md
Comment (Oberon00) [reply text/0134-remove-span-status.md:66]: Do we expect code behind new OTLP receivers that don't read (deprecated_)status anymore to still need status?

File: text/0134-remove-span-status.md
Comment (Oberon00) [reply text/0134-remove-span-status.md:43]: Make sense.

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:65]: I think having message without code will be confusing, because current state is exactly the opposite: you may have code without message.

File: text/0134-remove-span-status.md
Comment (iNikem) [reply text/0134-remove-span-status.md:66]: Don't know, that is way we leave it there for the time being. Collector maintainers may decide when they want to remove this translation.

File: text/0134-remove-span-status.md
Comment (Oberon00) [reply text/0134-remove-span-status.md:65]: But you may also have a message with an OK code. If you are worried about confusion, then set the status code additionally when the message is non-empty.

File: text/0134-remove-span-status.md
Comment (carlosalberto) [reply text/0134-remove-span-status.md:87]: @tigrannajaryan It seems open-telemetry/opentelemetry-specification#822 will be effectively merged soon (the `error.hint` Span attribute at least). Would that be enough for the Collector?

**Comment by carlosalberto** [comment]
@open-telemetry/specs-approvers Please review this OTEP. We need one more approval to proceed (and it's possible some of you haven't had time to pay attention to check this OTEP out ;) )

**Review by arminru — APPROVED** [review]

**Review by Oberon00 — APPROVED** [review]

File: text/0134-remove-span-status.md
Comment (tigrannajaryan) [reply text/0134-remove-span-status.md:87]: I am reluctant to give an approval until I clearly understand what we replace this by.

If https://github.com/open-telemetry/opentelemetry-specification/pull/822 is the replacement can we wait for it be merged first and update this OTEP to make use of it? It is an open question in this OTEP and is one of the central problems that this OTEPs needs to solve, but remains partially unsolved for now.

Comment (tigrannajaryan) [reply text/0134-remove-span-status.md:87]: @iNikem To be clear, I am not opposed to this OTEP, but I think it could be improved and an open question can be hopefully closed if https://github.com/open-telemetry/opentelemetry-specification/pull/822 is accepted.

We do not typically alter OTEPs after they are merged, so I don't know if a follow-up PR that would make the improvement is a reasonable approach. Because of this I also think that it may make sense to wait for https://github.com/open-telemetry/opentelemetry-specification/pull/822 and then make any required changes to this OTEP.

If you do not think that https://github.com/open-telemetry/opentelemetry-specification/pull/822 can have any significant impact on this OTEP then we don't need to wait for it and can assume https://github.com/open-telemetry/opentelemetry-specification/pull/822 does not exist. However in that case the way I perceive this PR changes a bit, because it seems to leave us without standardized error reporting approach and I think we need such approach.

Comment (bogdandrutu) [reply text/0134-remove-span-status.md:87]: I think the otep may change as a result of resolving any open question, but that doesn't mean we should stop this. I think the main point of this otep is the removing of span status and an action item is required to think about if we need an equivalent of `error.hint` or not separately.

Trying to make progress so we don't get stuck, but I can see your point as well, if you suggest that we really need a way to signal error from the application (which I don't believe we need) then I understand why we should block this until we have that.

Comment (iNikem) [reply text/0134-remove-span-status.md:87]: @tigrannajaryan can you express your opinion in open-telemetry/opentelemetry-specification#822? If you think that it may resolve your hesitation about this PR

Comment (bogdandrutu) [reply text/0134-remove-span-status.md:87]: > I am not against moving forward with this OTEP, but I believe it will have an important open question and will be unactionable in the Collector if it does not take into account either the existence of "error.hint" or the lack of whatever else needs to replace Span Status. IMO, it will mean we will need another OTEP to correct this OTEP once "error.hint" is merged or rejected.

Not sure you need an OTEP for that, that is a semantic convention, and collector will use it.

Comment (tigrannajaryan) [reply text/0134-remove-span-status.md:87]: @bogdandrutu sure, that probably works too.

**Comment by tigrannajaryan** [comment]
All, I don't want to be the person who blocks this OTEP. My only worry is that "error.hint" PR is not going to be merged due to disagreements and we will remove the Span status without having a reasonable replacement.

If everyone else in this thread thinks that it is not a problem and we will find a solution to that I am happy to approve and go ahead with this OTEP.

**Comment by tedsuo** [comment]
@iNikem  Now that [OTEP 136](https://github.com/open-telemetry/oteps/blob/master/text/trace/0136-error_flagging.md) is merged, this PR may be closed.

**Comment by iNikem** [comment]
Closed in favor of #136

**Pull request closed without merging**
